src/scripts/clipperUI/clipper.tsx

```diff
@@ -620,7 +622,18 @@ class ClipperClass extends ComponentBase<ClipperState, {}> {
 		return signOutState;
 	}
 
-	private handleStartClip() {
+	private handleStartClip(): void {
+		const pdfPreviewInfo = this.state.pdfPreviewInfo;
+		if (this.state.currentMode.get() === ClipMode.Pdf && !pdfPreviewInfo.allPages) {
+			const parsePageRangeOperation = StringUtils.parsePageRange(pdfPreviewInfo.selectedPageRange, this.state.pdfResult.data.get().pdf.numPages());
```

Comment:
That seems fine with me. It makes conceptual sense as well. The logic should live in the lowest component it can. The top level `Clipper` knows how to save the Clip, but it is the interaction between the clipButton and the options panel that gates it, not necessarily something at the `Clipper` level.